swo/swosync/logicalsync.go

```diff
@@ -40,7 +40,7 @@ func (l *LogicalReplicator) doSync(ctx context.Context, final bool) error {
 	if err != nil {
 		return fmt.Errorf("read changes: begin tx: %w", err)
 	}
-	defer l.srcConn.Exec(ctx, `rollback`)
+	defer func(l *LogicalReplicator) { _, _ = l.srcConn.Exec(ctx, `rollback`) }(l)
```

Comment:
Pass in the connection instead of `LogicalReplicator`; that way, we don't need to comb through every source file to ensure `srcConn` can't get reassigned. The most crucial part is ensuring we issue the rollback on the correct connection.

Reply:
done